irods-vfs-impl/src/main/java/org/irods/nfsrods/vfs/IRODSVirtualFileSystem.java

```diff
@@ -832,7 +839,57 @@ public boolean hasIOLayout(Inode _inode) throws IOException
     @Override
     public Inode link(Inode _parent, Inode _existing, String _target, Subject _subject) throws IOException
     {
-        throw new UnsupportedOperationException("Not supported");
+        log_.debug("vfs::link");
+
+        Path linkParentPath = getPath(toInodeNumber(_parent));
+        Path linkTargetPath = getPath(toInodeNumber(_existing));
+        Path linkNamePath = linkParentPath.resolve(_target);
+
+        log_.debug("list - _parent   = {}", linkParentPath);            // The parent directory of _target.
+        log_.debug("list - _existing = {}", linkTargetPath);            // The file or directory to link to.
+        log_.debug("list - _target   = {}", _target);                   // The filename or absolute path of the link.
+        log_.debug("list - _subject  = {}", Subjects.getUid(_subject)); // The UID of the user who executed this command.
+        
+        // _parent + _target => the absolute path of the hard link.
+        
+        Map<String, String> jsonInput = new HashMap<>();
+        jsonInput.put("operation", "hard_link_create");
+        jsonInput.put("logical_path", linkTargetPath.toString());
+        jsonInput.put("replica_number", "0");
```

Comment:
How do we want to handle cases where replica zero does not exist?

Reply:
This could be controlled via the following:
- configuration
- metadata
- genquery that selects a specific replica based on certain criteria provided by the admin

Reply:
i don't know how we could safely assume anything... doesn't it have to come from the catalog?

Reply:
Maybe it executes genquery to find the latest good replica and it creates the hard link to that?

Reply:
I think latest good replica is the answer... trying to figure out how to get that information without another trip to the catalog...